src/components/Pagination/index.tsx

```diff
@@ -1,26 +1,27 @@
+import { useState } from "react";
 import { Button, Icon, Input } from "@stellar/design-system";
 import "./styles.scss";
 
 interface PaginationProps {
   currentPage: number;
   maxPages: number;
-  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
-  onBlur: (currentPage: number) => void;
-  onPrevious: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
-  onNext: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+  onSetPage: (page: number) => void;
   isLoading: boolean;
 }
 
 export const Pagination = ({
   currentPage,
   maxPages,
-  onChange,
-  onBlur,
-  onPrevious,
-  onNext,
+  onSetPage,
   isLoading,
 }: PaginationProps) => {
-  const isError = currentPage > maxPages;
+  const [page, setPage] = useState<number | undefined>();
```

Comment:
Using this local `page` state variable to keep manual page input change local and trigger parent component update only on blur.